dist/scripts/binary-dedupe.sh

```diff
@@ -75,3 +75,43 @@ mv "$SPARK3XX_COMMON_DIR" $PARALLEL_WORLDS_DIR/
 #  spark30x-common
 #  spark31x-common
 #  spark32x-common
+
+# Verify that all class files in the conventional jar location are bitwise
+# identical regardless of the Spark-version-specific jar.
+#
+# At this point the duplicate classes have not been removed from version-specific jar
+# locations such as parallel-world/spark312.
+# For each unshimmed class file look for all of its copies inside /spark3* and
+# and count the number of distinct checksums. There are two representative cases
+# 1) The class is contributed to the unshimmed location via the unshimmed-from-each-spark3xx list. These are classes
+#    carrying the shim classifier in their package name such as
+#    com.nvidia.spark.rapids.spark312.RapidsShuffleManager. They are by unique by construction,
+#    and will have zero copies in any non-spark312 shims. Although such classes are currently excluded from
+#    being copied to the /spark312 Parallel World we keep the algorithm below general without assuming this.
+#
+# 2) The class is contributed to the unshimmed location via unshimmed-common. These are classes that
+#    that have the same package and class name across all parallel worlds.
+#
+#  So if the number of distinct class files per class in the unshimmed location is < 2, the jar
+#  is content is as expected
+#
+#  If we find an unshimmed class file occurring > 1  we fail the build and the code must be refactored
+#  until bitwise-identity of each unshimmed class is restored.
+
+# Determine the list of unshimmed class files
+UNSHIMMED_LIST_TXT=unshimmed-result.txt
+find . -name '*.class' -not -path './'$PARALLEL_WORLDS_DIR/'spark*' | \
```

Comment:
I'm confused by this, how is this looking at unshimmed classes, this looks like its looking at all classes in the shims?  How is this not failing on the ProxyShuffle class that we were grabbing from 3.0?  I must be missing something basic here

Reply:
On L103 we look for all classes that are *not* in /spark*  `-not -path './'$PARALLEL_WORLDS_DIR/'spark*'` 
then for all those unshimmed classes  (including `org/apache/spark/sql/rapids/ProxyRapidsShuffleInternalManagerBase.class`) 

then using find rooted off `target` (probably should tighten to `target/parallel-world` ) we find generally all copies of every unshimmed class.
```
find . -path "./*/org/apache/spark/sql/rapids/ProxyRapidsShuffleInternalManagerBase.class"
```
we take md5sum of every copy and "count distinct" the checksums. If our assumption about using 301 for all shims is right, the count is 1, if > 1 the assumption was wrong and we fail the build.

Reply:
my mistake, I missed the -not in here